Utils/Event Pump.cpp

```diff
@@ -1237,17 +1389,16 @@ BOOLEAN ExecuteGameEvent( EVENT *pEvent )
 BOOLEAN ClearEventQueue( void )
 {
 	// clear out the event queue
-	EVENT					*pEvent;
+	EVENT pEvent;
 	while( EventQueueSize( PRIMARY_EVENT_QUEUE ) > 0 )
 	{
 		// Get Event
-		if ( RemoveEvent( &pEvent, 0, PRIMARY_EVENT_QUEUE) == FALSE )
+		if (PopEvent( &pEvent, PRIMARY_EVENT_QUEUE) == FALSE )
 		{
 			return( FALSE );
 		}
+		FreeEvent(&pEvent);

```

Comment:
This is going to call `free()` on something that wasn't either malloc'd or new'd in the first place, I don't think this is right

Reply:
It's currently only freeing the pData that is malloc'd. The EVENT itself is not freed as they are not malloc'd in the code. Poor naming, I know.